sql/2019/12_Mobile_Web/12_06.sql

```diff
@@ -1,35 +1,35 @@
 #standardSQL
 
-# breakpoint under 600px width
-# @issue: how to group by page domain/url?
+# % of pages that include a stylesheet with a breakpoint under 600px.
+# (!) 7.71 TB
 
 CREATE TEMPORARY FUNCTION hasBreakpoint(css STRING)
 RETURNS BOOLEAN LANGUAGE js AS '''
+function matchAll(re, str) {
+  var results = [];
+  while ((matches = re.exec(str)) !== null) {
+    results.push(matches[1]);
+  }
+  return results;
+}
 
 try {
-  var reduceValues = (values, rule) => {
-    if (rule.type == 'media') {
-      var minValue = rule.media.matchAll(/max-width: ([0-9]+)px/i);
-      var maxValue = rule.media.matchAll(/min-width: ([0-9]+)px/i);
-      if(minValue[0] < 600 || maxValue[0] < 600) {
-        return true
-      }
-
-      return reduceValues(values, rule.rules);
-    }
-  };
-
   var $ = JSON.parse(css);
-  return $.stylesheet.rules.reduce(reduceValues, []);
-
+  return $.stylesheet.rules.some(rule => {
+    return rule.type == 'media' &&
+        rule.media &&
+        matchAll(/width:\\s*(\\d+)px/ig, rule.media).some(match => +match < 600);
+  });
 } catch (e) {
-  return [];
+  false;
 }
 ''';
 
-#SELECT
-#    COUNT(0) as count,
-#    COUNTIF(hasBreakpoint(payload)) AS occurence,
-#    ROUND(COUNTIF(hasBreakpoint(payload)) * 100 / SUM(COUNT(0)) OVER (), 2) AS occurence_perc
-#FROM
-#    `httparchive.almanac.parsed_css`
+SELECT
+  COUNT(DISTINCT page) AS pages,
+  ROUND(COUNT(DISTINCT page) * 100 / (SELECT COUNT(0) AS total FROM `httparchive.summary_pages.2019_07_01_mobile`), 2) AS pct
```

Comment:
Because of the `WHERE hasBreakpoint` the `COUNT(0)` will only include pages with breakpoints, not all pages.

Reply:
Are you sure? The COUNT(0) is in a sub-select out of the`WHERE hasBreakpoint ` restriction.